src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -19,20 +28,22 @@ public class StakeValidator
         /// <summary>Time interval in minutes that is used in the retarget calculation.</summary>
         private const int RetargetIntervalMinutes = 16;
 
-        /// <summary>Instance logger.</summary>
-        private readonly ILogger logger;
-
         /// <summary>Class logger.</summary>
         private static readonly ILogger clogger;
 
-        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
-        private readonly Network network;
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
 
         /// <summary>Database of stake related data for the current blockchain.</summary>
         private readonly StakeChain stakeChain;
 
+        /// <summary>Chain of headers.</summary>
```

Comment:
we don't have this one described well in the code base, so i'd probably be most happy about

`<summary>Thread safe access to the best chain of block headers from genesis.</summary>`

- it is important for me to know it is THE best chain the node is aware of
- it is supposed to be thread safe 
- goes down to the genesis